test/CowWrapper.t.sol

```diff
@@ -0,0 +1,293 @@
+        solvers[solver] = true;
+    }
+
+    function isSolver(address solver) external view returns (bool) {
+        return solvers[solver];
+    }
+}
+
+contract MockSettlement {
+    struct SettleCall {
+        IERC20[] tokens;
+        uint256[] clearingPrices;
+        GPv2Trade.Data[] trades;
+        bytes additionalData;
+    }
+
+    SettleCall[] public settleCalls;
+    CowAuthentication private immutable _authenticator;
+
+    constructor(CowAuthentication authenticator_) {
+        _authenticator = authenticator_;
+    }
+
+    function authenticator() external view returns (CowAuthentication) {
+        return _authenticator;
+    }
+
+    function settle(
+        IERC20[] calldata tokens,
+        uint256[] calldata clearingPrices,
+        GPv2Trade.Data[] calldata trades,
+        GPv2Interaction.Data[][3] calldata
+    ) external {
+        SettleCall storage call_ = settleCalls.push();
+        call_.tokens = tokens;
+        call_.clearingPrices = clearingPrices;
+
+        for (uint256 i = 0; i < trades.length; i++) {
+            call_.trades.push(trades[i]);
+        }
+
+        // Extract any additional data appended after standard calldata
+        uint256 expectedLength = 4 + 4 * 32;
+        expectedLength += 32 + tokens.length * 32;
+        expectedLength += 32 + clearingPrices.length * 32;
+        expectedLength += 32;
+        for (uint256 i = 0; i < trades.length; i++) {
+            expectedLength += 9 * 32 + 32 + trades[i].signature.length;
+        }
+        expectedLength += 32;
+        expectedLength += 3 * 32;
+        expectedLength += 3 * 32;
+
+        if (msg.data.length > expectedLength) {
+            call_.additionalData = msg.data[expectedLength:];
+        }
+    }
+
+    function getSettleCallCount() external view returns (uint256) {
+        return settleCalls.length;
+    }
+
+    function getLastSettleCall()
+        external
+        view
+        returns (uint256 tokenCount, uint256 priceCount, uint256 tradeCount, bytes memory additionalData)
+    {
+        require(settleCalls.length > 0, "No settle calls");
+        SettleCall storage lastCall = settleCalls[settleCalls.length - 1];
+        return (lastCall.tokens.length, lastCall.clearingPrices.length, lastCall.trades.length, lastCall.additionalData);
+    }
+}
+
+// Test wrapper that exposes internal functions
+contract TestWrapper is CowWrapper {
+    string public constant name = "Test Wrapper";
+
+    // Track _wrap calls
+    struct WrapCall {
+        bytes settleData;
+        bytes wrapperData;
+    }
+
+    WrapCall[] public wrapCalls;
+
+    constructor(CowSettlement settlement_) CowWrapper(settlement_) {}
+
+    function _wrap(bytes calldata settleData, bytes calldata wrapperData, bytes calldata remainingWrapperData)
+        internal
+        override
+    {
+        // Record the wrap call
+        WrapCall storage call_ = wrapCalls.push();
+        call_.settleData = settleData;
+        call_.wrapperData = wrapperData;
```

Comment:
I thought foundry had already a way to get this calls without needing you to save it in storage